pkg/generate/app/cmd/newapp.go

```diff
@@ -222,27 +224,58 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 // AddArguments converts command line arguments into the appropriate bucket based on what they look like
 func (c *AppConfig) AddArguments(args []string) []string {
 	unknown := []string{}
+	c.ArgumentClassificationErrors = map[string]error{}
 	for _, s := range args {
-		switch {
-		case cmdutil.IsEnvironmentArgument(s):
+		if len(s) == 0 {
+			continue
+		}
+
+		validated := false
+
+		// had to move off of elegant switch to capture errors for future display, but mimic
+		// order of precedence established by switch on env arg, then source repo, then comp
+		// ref, then possible template, with no fallthroughs and updates to c.Components or
+		// c.SourceRepositories if a classification of higher precedence succeeds
+
+		rc := cmdutil.IsEnvironmentArgument(s)
+		if rc {
 			glog.V(2).Infof("treating %s as possible environment argument\n", s)
 			c.Environment = append(c.Environment, s)
-		case app.IsPossibleSourceRepository(s):
+			validated = true
+		}
+
+		remote, err := app.IsRemoteRepository(s)
+		c.ArgumentClassificationErrors["Remote access"] = err
+		local, err := app.IsDirectory(s)
+		c.ArgumentClassificationErrors["Local access"] = err
```

Comment:
I'm not sure how the additional information is useful since there's nothing the user can do about it (eg if we think it's an environment variable, it doesn't really matter if we also think it's a source repo, or are sure it's not a source repo because of error X, it's still going to get processed/consumed as an environment variable).

If anything it may just lead to more confusion since anytime new-app fails overall(for unrelated reasons), it's going to print out messages talking about the error it got when trying to treat an env variable argument as a source repo.